cloudbuild.yaml

```diff
@@ -0,0 +1,162 @@
+substitutions:
+  _REGISTRY: "us-central1-docker.pkg.dev"
+  _REPO: "seda-data-proxy"
+  _PROJECT: "${PROJECT_ID}"
+  _IMAGE_NAME: "seda-data-proxy"
+
+options:
+  machineType: 'E2_HIGHCPU_8'
+  logging: CLOUD_LOGGING_ONLY
+
+steps:
+#
+# Step 1: Configure Docker Auth
+#
+- name: 'gcr.io/cloud-builders/gcloud'
+  id: 'auth'
+  args:
+    - 'auth'
+    - 'configure-docker'
+    - '${_REGISTRY}'
+    - '--quiet'
+
+#
+# Step 2: Set up QEMU for Multi-Arch Builds
+#
+- name: 'gcr.io/cloud-builders/docker'
+  id: 'setup-qemu'
+  args:
+    - 'run'
+    - '--privileged'
+    - 'multiarch/qemu-user-static'
+    - '--reset'
+    - '-p'
+    - 'yes'
+
+#
+# Step 3: Build and Push AMD64 Image
+#
+- name: 'gcr.io/cloud-builders/docker'
+  id: 'build-amd64'
+  args:
+    - 'build'
+    - '--platform'
+    - 'linux/amd64'
+    - '--build-arg'
+    - 'TARGET_ARCH=bun-linux-x64'
+    - '-t'
+    - '${_REGISTRY}/${_PROJECT}/${_REPO}/${_IMAGE_NAME}:${TAG_NAME}-amd64'
+    - '-f'
+    - '.build/docker/Dockerfile'
+    - '.'
+
+- name: 'gcr.io/cloud-builders/docker'
+  id: 'push-amd64'
+  waitFor: ['auth', 'build-amd64']
+  args:
+    - 'push'
+    - '${_REGISTRY}/${_PROJECT}/${_REPO}/${_IMAGE_NAME}:${TAG_NAME}-amd64'
+
+#
+# Step 4: Build and Push ARM64 Image
+#
+- name: 'gcr.io/cloud-builders/docker'
+  id: 'build-arm64'
+  args:
+    - 'build'
+    - '--platform'
+    - 'linux/arm64'
+    - '--build-arg'
+    - 'TARGET_ARCH=bun-linux-arm64'
+    - '-t'
+    - '${_REGISTRY}/${_PROJECT}/${_REPO}/${_IMAGE_NAME}:${TAG_NAME}-arm64'
+    - '-f'
+    - '.build/docker/Dockerfile'
+    - '.'
+
+- name: 'gcr.io/cloud-builders/docker'
+  id: 'push-arm64'
+  waitFor: ['auth', 'build-arm64']
+  args:
+    - 'push'
+    - '${_REGISTRY}/${_PROJECT}/${_REPO}/${_IMAGE_NAME}:${TAG_NAME}-arm64'
+
+#
+# Step 5: Create and Push Manifest List
+#
+- name: 'gcr.io/cloud-builders/docker'
+  id: 'create-manifest'
+  waitFor: ['push-amd64', 'push-arm64']
+  entrypoint: 'bash'
+  args:
+    - '-c'
+    - |
+      docker manifest create \
+        ${_REGISTRY}/${_PROJECT}/${_REPO}/${_IMAGE_NAME}:${TAG_NAME} \
+        --amend ${_REGISTRY}/${_PROJECT}/${_REPO}/${_IMAGE_NAME}:${TAG_NAME}-amd64 \
+        --amend ${_REGISTRY}/${_PROJECT}/${_REPO}/${_IMAGE_NAME}:${TAG_NAME}-arm64
+      docker manifest push ${_REGISTRY}/${_PROJECT}/${_REPO}/${_IMAGE_NAME}:${TAG_NAME}
+
+#
+# Step 6: Create GitHub Release
+#
+- name: 'node:18'
```

Comment:
Should we not use a version that’s still in the maintenance window?

Reply:
let's replace it to `node:20`